### PR TITLE
Add a sort_posts function

### PR DESCRIPTION
**Pull request by Kwpolska**
This is #2602 — it adds a `sort_posts` function (available as Jinja filter in global context), which allows general-purpose timeline sorting. Example usage:

```python
sort_posts(timeline, 'title', 'date')
sort_posts(timeline, 'author', reverse=True)
```

cc @liffiton — does this match your needs?

### CODE REVIEW

**Comment by felixfontein** [comment]
From `sort_posts(timeline, 'title', 'date')` I'd assume that the posts are sorted by title, and if they have the same title, then they are sorted by date. But the code actually sorts them the other way around: first by date, and only for posts on the same date (i.e. with identical timestamp, as date means timestamp) by title.

**Comment by Kwpolska** [comment]
That’s a sane assumption to make, I added a little `reversed()` call to account for this.

**Comment by felixfontein** [comment]
How about being able to sort some keys reversed, and some not, by prepending `reverse_` to the key name? Standard use case is sorting by date and title: you might want to sort by date reversed to get the newest posts first, but by title alphabetically.

**Comment by Kwpolska** [comment]
@felixfontein done — prepend with `-` to get a key reversed (same syntax as used in Django). Review welcome.

File: nikola/utils.py [hunk 1]
```diff
@@ -1947,6 +1950,35 @@ def clean_before_deployment(site):
...
+    """
+    # We reverse the keys to get the usual ordering method: the first key
+    # provided is the most important sorting predicate (first by 'title', then
+    # by 'date' in the first example)
+    for key in reversed(keys):
+        try:
+            # An attribute (or method) of the Post object
+            a = getattr(posts[0], key)
+            if callable(a):
+                keyfunc = operator.methodcaller(key)
```
Comment (felixfontein) [inline nikola/utils.py]: Does this actually work when `key[0] == '-'`?

Comment (Kwpolska) [inline nikola/utils.py]: Oh, right. Fixed.

**Pull request merged**